### PR TITLE
Use button functionality to make some fields clickable (now including `crossref`)

### PR DESCRIPTION
**Pull request by Hugo-Heagren**
Replaces #264.

Use Emacs-standard button library to make some text clickable, replacing the previous mechanism. Reasons are given in the commit messages and #264. Make `crossref` fields clickable, to follow the crossref (similar to opening a file).

### CODE REVIEW

File: ebib-utils.el [hunk 1]
```diff
@@ -2387,14 +2387,16 @@ are returned as a string."
...
-    (if (and str (string-match-p "^[0-9]" str))
-        (setq str (concat "https://dx.doi.org/" str)))
-    (if str
-        (propertize "www"
-                    'face '(:height 0.8 :inherit ebib-link-face)
-                    'mouse-face 'highlight
-                    'help-echo str)
+  (let* ((val (ebib-get-field-value field key db 'noerror 'unbraced 'xref))
+	 ;; Unless field is doi, assume the string is a full url
+	 (str (if (case-fold-string= field "doi")
```
Comment (joostkremers) [inline ebib-utils.el:2392]: Note that the DOI field may already start with `https://dx.doi.org/`, which is why the original code only adds it if the field value starts with a number. (DOIs tend to start with `10`, though I admit I'm not sure if that's mandatory. A better test would be to see if the field value starts with `https`.)

Comment (Hugo-Heagren) [reply ebib-utils.el]: Thanks for pointing this out. Fixed and pushed. Turns out, DOIs [*should* always start with `10`](https://www.doi.org/doi_handbook/2_Numbering.html#2.2.2), though I don't know whether this is actually enforced, so I've stuck with the check for a single leading number 0-9.

**Comment by joostkremers** [comment]
I get a bunch of compiler warnings trying to compile this PR. Among others, they include a bunch of these:

```
ebib-utils.el:2397:10: Warning: button-buttonize called with 4 arguments, but
    accepts only 2-3
```

I'm running an Emacs pretest version (28.1.91), so perhaps this will change in Emacs 29, but for now, `button-buttonize` takes three arguments, the third one optional. The third argument should be a list, I assume:

``` emacs-lisp
        (button-buttonize "www"
		          'ebib--call-browser ;; callback
		          (list str           ;; data (passed to callback)
		                str))         ;; help-echo

```

I'm also adding a few inline comments.

File: ebib-utils.el [hunk 2]
```diff
@@ -2387,14 +2387,17 @@ are returned as a string."
...
-    (if (and str (string-match-p "^[0-9]" str))
-        (setq str (concat "https://dx.doi.org/" str)))
-    (if str
-        (propertize "www"
-                    'face '(:height 0.8 :inherit ebib-link-face)
-                    'mouse-face 'highlight
-                    'help-echo str)
+  (let* ((val (ebib-get-field-value field key db 'noerror 'unbraced 'xref))
+	 ;; Unless field is doi, assume the string is a full url
+	 (str (if (and (case-fold-string= field "doi")
```
Comment (joostkremers) [inline ebib-utils.el:2392]: `case-fold-string=` isn't part of Emacs. It's actually given as an example in the Emacs manual `(info "(elisp) Defining Hash")`, perhaps you got it from there?

**Comment by joostkremers** [comment]
> I'm running an Emacs pretest version (28.1.91), so perhaps this will change in Emacs 29, but for now, `button-buttonize` takes three arguments, the third one optional. The third argument should be a list, I assume:

That doesn't make sense, actually... If the fourth argument is a help echo for the button, it would be weird to pass it to the callback function.

File: ebib-utils.el [hunk 3]
```diff
@@ -2387,14 +2387,17 @@ are returned as a string."
...
+		       (string-match-p "^[0-9]" str))
```
[the remark this answers is not in the document]
Comment (Hugo-Heagren) [reply ebib-utils.el:2393]: Think you're right -- fixed.

File: ebib-utils.el (hunk 2 above)
Comment (Hugo-Heagren) [reply ebib-utils.el:2392]: Strange -- I checked and it actually comes from EMMS (which I use, so I have it loaded). EMMS doesn't prefix the function with a proper name, so I assumes it was part of Emacs itself. Fixed.

**Comment by Hugo-Heagren** [comment]
As far as I can see then:
- `buttonize` is the most current, 'correct' way, but is too recent to be available on most people's machines
- `button-buttonize` is the (now obsolete) previously 'correct' way, was [committed in December of 2020](https://git.savannah.gnu.org/cgit/emacs.git/commit/lisp/button.el?id=14ffab8263eb219fe0c49ad4e0a3476316c542c0), so is too new to be in 26.1 or even 27.1 (released in August of 2020).

So I'll just add the properties 'manually' with `propertize`. This also completely removes the problem of which versions of the function have which arguments.

**Comment by Hugo-Heagren** [comment]
> So I'll just add the properties 'manually' with `propertize`. This also completely removes the problem of which versions of the function have which arguments.

Done, and pushed the new versions of the relevant commits.

**Comment by joostkremers** [comment]
Hey, it took a bit of time for me to take a look at this more closely. Looks good, except one small thing: The notes symbol in the index buffer was also clickable: it was handled by `ebib-index-open-at-point`. You'll need to give the function `ebib-notes-display-note-symbol` the same treatment as `ebib-display-www-link`. Or wait until I do it. :smile:

**Comment by Hugo-Heagren** [comment]
> You'll need to give the function `ebib-notes-display-note-symbol` the same treatment as `ebib-display-www-link`

Done!

**Comment by joostkremers** [comment]
Finally got to take another look and merged this. Thanks!

**Pull request merged**